src/ManifoldsBase.jl

```diff
@@ -440,34 +440,110 @@ Compute the inner product of tangent vectors `X` and `Y` at point `p` from the
-    isapprox(M::AbstractManifold, p, q; kwargs...)
+    isapprox(M::AbstractManifold, p, q; error::Symbol=none, kwargs...)
 
 Check if points `p` and `q` from [`AbstractManifold`](@ref) `M` are approximately equal.
 
+The keyword argument can be used to get more information for the case that
+the result is false, if the concrete manifold provides such information.
+Currently the following are supported
+* `:error` - throws an error if `isapprox` evaluates to false, providing possibly a more detailed error.
+  Note that this turns `isapprox` basically to an `@assert`.
+* `:info` – prints the information in an `@info`
+* `:warning` – prints the information in an `@warn`
+* `:none` (default) – the function just returns `true`/`false`
+
 Keyword arguments can be used to specify tolerances.
 """
-isapprox(::AbstractManifold, x, y; kwargs...) = isapprox(x, y; kwargs...)
+function isapprox(M::AbstractManifold, p, q; error::Symbol = :none, kwargs...)
+    ma = check_approx(M, p, q; kwargs...)
+    if ma !== nothing
+        (error === :error) && throw(ma)
+        if isnan(ma.val)
+            s = "$(typeof(ma))\n$(ma.msg)"
+        else
+            s = "$(typeof(ma)) with $(ma.val)\n$(ma.msg)"
+        end
+        (error === :info) && @info s
+        (error === :warning) && @warn s
+        return false
+    end
+    return true
+end
+
+function check_approx(M::AbstractManifold, p, q; kwargs...)
+    # fall back to classical approx mode - just that we do not have a reason then
+    res = isapprox(p, q; kwargs...)
+    # since we can not assume distance to be implemented, we can not provide a default value
+    res && return nothing
+    s = "The two points $p and $q on $M are not (approximately) equal."
+    return ApproximatelyError(s)
+end
+
 
 """
-    isapprox(M::AbstractManifold, p, X, Y; kwargs...)
+    isapprox(M::AbstractManifold, p, X, Y; error:Symbol=:none; kwargs...)
 
 Check if vectors `X` and `Y` tangent at `p` from [`AbstractManifold`](@ref) `M` are approximately
 equal.
 
+The optional positional argument can be used to get more information for the case that
+the result is false, if the concrete manifold provides such information.
+Currently the following are supported
+
+* `:error` - throws an error if `isapprox` evaluates to false, providing possibly a more detailed error.
+  Note that this turns `isapprox` basically to an `@assert`.
+* `:info` – prints the information in an `@info`
+* `:warning` – prints the information in an `@warn`
+* `:none` (default) – the function just returns `true`/`false`
+
+By default these informations are collected by calling [`check_approx`](@ref).
+
 Keyword arguments can be used to specify tolerances.
 """
-isapprox(::AbstractManifold, p, X, Y; kwargs...) = isapprox(X, Y; kwargs...)
+function isapprox(M::AbstractManifold, p, X, Y; error::Symbol = :none, kwargs...)
+    mat = check_approx(M, p, X, Y; kwargs...)
+    if mat !== nothing
+        (error === :error) && throw(mat)
+        if isnan(mat.val)
```

Comment:
I think we could always print `val` even if it's `NaN` for consistency?

Reply:
My fear is that the nan might confuse a used

Reply:
It might, but so might lack of any number. How about changing the message to something like `"$(typeof(ma)) (value: $(ma.val))\n$(ma.msg)"`?

Reply:
Then I still find `Value: NaN` irritating to the user, since it is actually not `NaN` but absence of information (at least information that can be provided).

Reply:
OK, then let's leave this as is.